core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -265,10 +271,15 @@ public static ElasticsearchException newConfigurationException(String processorT
         return processors;
     }
 
-    public static TemplateService.Template compileTemplate(String processorType, String processorTag, String propertyName,
-                                                           String propertyValue, TemplateService templateService) {
+    public static TemplateScript.Factory compileTemplate(String processorType, String processorTag, String propertyName,
+                                                           String propertyValue, ScriptService scriptService) {
         try {
-            return templateService.compile(propertyValue);
+            if (scriptService.isLangSupported(DEFAULT_TEMPLATE_LANG)) {
+                Script script = new Script(ScriptType.INLINE, DEFAULT_TEMPLATE_LANG, propertyValue, Collections.emptyMap());
+                return scriptService.compile(script, TemplateScript.CONTEXT);
+            } else {
+                return (params) -> () -> propertyValue;
```

Comment:
I don't like this but I understand why it is here. Can you please add a comment explaining that this is for rest tests when using the integ-test-zip?